### PR TITLE
Use SPI to initialize JobErrorHandler in test cases #1515/Use mock & verify for EmailJobErrorHandlerTest #1517

### LINKED ISSUE #1515: Use SPI to initialize JobErrorHandler in test cases
- [ ] IgnoreJobErrorHandlerTest
- [ ] LogJobErrorHandlerTest
- [ ] ThrowJobErrorHandlerTest
- [ ] EmailJobErrorHandlerTest
- [ ] WechatJobErrorHandler
- [ ] DingtalkJobErrorHandlerTest

### LINKED ISSUE #1517: Use mock & verify for EmailJobErrorHandlerTest
Please mock and verify `session` for EmailJobErrorHandlerTest

### PR DESCRIPTION
**Pull request by jiang2015**
Fixes #1515 .
Fixes #1517  .

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-wechat/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatJobErrorHandlerTest.java [hunk 1]
```diff
@@ -108,6 +110,15 @@ private String getHost() {
         return String.format("http://%s:%s", HOST, PORT);
     }
     
+    private WechatJobErrorHandler getWechatJobErrorHandler() {
+        for (JobErrorHandler jobErrorHandler : ServiceLoader.load(JobErrorHandler.class)) {
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-wechat/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatJobErrorHandlerTest.java]: jobErrorHandler -> each

File: elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-email/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/email/EmailJobErrorHandlerTest.java [hunk 2]
```diff
@@ -50,4 +53,28 @@ public void assertHandleExceptionWithYAMLConfiguration() throws ReflectiveOperat
...
+        assertNotNull(session);
+        assertThat(session.getProperties().get("mail.smtp.host"), equalTo("yaml.email.com"));
+        assertThat(session.getProperties().get("mail.debug"), equalTo("true"));
+        assertThat(session.getProperties().get("mail.smtp.port"), equalTo(123));
+        assertThat(session.getProperties().get("mail.transport.protocol"), equalTo("yaml.smtp"));
+        assertThat(session.getProperties().get("mail.smtp.auth"), equalTo("true"));
+    }
+    
+    private EmailJobErrorHandler getEmailJobErrorHandler() {
+        for (JobErrorHandler jobErrorHandler : ServiceLoader.load(JobErrorHandler.class)) {
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-email/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/email/EmailJobErrorHandlerTest.java]: rename jobErrorHandler -> each

File: elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-dingtalk/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/dingtalk/DingtalkJobErrorHandlerTest.java [hunk 3]
```diff
@@ -111,4 +113,13 @@ public static void close() {
             restfulService.shutdown();
         }
     }
+    
+    private DingtalkJobErrorHandler getDingtalkJobErrorHandler() {
+        for (JobErrorHandler jobErrorHandler : ServiceLoader.load(JobErrorHandler.class)) {
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-dingtalk/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/dingtalk/DingtalkJobErrorHandlerTest.java]: jobErrorHandler -> each

File: elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-wechat/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatJobErrorHandlerTest.java (hunk 1 above)
Comment (jiang2015) [reply elasticjob-error-handler/elasticjob-error-handler-impl/elasticjob-error-handler-wechat/src/test/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatJobErrorHandlerTest.java]: more modified

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**